src/main/java/homer/model/outlets/Outlet.java

```diff
@@ -137,16 +87,26 @@ public Optional<Device<?>> getDevice() {
     public final void updateTick(final Duration deltaTime) {
         final double defaultMaxPower = 150.0;
         final double defaultRandomIncrement = Math.random() * 10 + 1;
-        double energy;
+        OutletState energy = new OutletState();
```

Comment:
Also this, or should I
`OutletState energy = (OutletState) this.getState()` ?

Reply:
Depends what is this supposed to do?

Reply:
Also you can save yourself the headache and keep the internal implementation untouched, technically there should only be changes in getState() and setState(), just creating and retrieving the appropriate DeviceState implementation for your needs.

Reply:
> Also you can save yourself the headache and keep the internal implementation untouched, technically there should only be changes in getState() and setState(), just creating and retrieving the appropriate DeviceState implementation for your needs.

100%, but let's not get too hung up on some private method's implementation. If observable behaviour is fine, and we don't have performance issues, it's good enough.